STL_Extension/include/CGAL/Uncertain.h

```diff
@@ -520,6 +494,69 @@ Uncertain<T> make_uncertain(Uncertain<T> t)
   return t;
 }
 
+// operator&& and operator|| are not provided because, unless their bool counterpart,
+// they lack the "short-circuiting" property.
+// We provide macros CGAL_AND and CGAL_OR, which attempt to emulate their behavior.
+// Key things : do not evaluate expressions twice, and evaluate the right hand side
+// expression only when needed.
+
+inline
+Uncertain<bool> CGAL_AND(const Uncertain<bool>& a, const Uncertain<bool>& b)
+{
+  return certainly_not(a) ? make_uncertain(false)
+                          : a & b;
+}
+
+inline
+Uncertain<bool> CGAL_AND(bool a, const Uncertain<bool>& b)
+{
+  return !a ? make_uncertain(false)
+            : a & b;
+}
```

Comment:
```suggestion
inline
Uncertain<bool> CGAL_AND(bool a, const Uncertain<bool>& b)
{
  return !a ? make_uncertain(false)
            : b;
}
```

Reply:
because `a` is `true`  in the `else` case no need to 'and' it.